src/v/rpc/rpc_compiler.py

```diff
@@ -144,16 +144,14 @@ class failure_probes;
     metrics::internal_metric_groups _metrics;
 };
 
-class {{service_name}}_client_protocol {
+class {{service_name}}_client_protocol {% if final_protocol %}final{% endif %} {
 public:
     explicit {{service_name}}_client_protocol(ss::lw_shared_ptr<::rpc::transport> t)
       : _transport(t) {
     }
 
-    virtual ~{{service_name}}_client_protocol() = default;
```

Comment:
is the reason for always omitting the virtual destructor here because the `final_protocol=False` case with subclasses where it might be desired aren't present in the existing test situations?

Reply:
@dotnwat - yes: just like in most cases we use subclassing today, there is no deletion through a virtual base pointer or anything like that: we use the fully derived type.

Specifically we use subclassing here:

https://github.com/redpanda-data/redpanda/blob/5485635b5aaa580fa45e0b275d0238659b19270e/src/v/rpc/transport.h#L458-L490

To create a "client" object that bundles together management of the transport + multiple RPC protocol objects, apparently as a convenience for testing.

I also considered just modifying the test not to do (e.g., to store the procotols as a tuple instead), but this seemed the path of least resistance.